src/make_trade.ts

```diff
@@ -36,27 +36,28 @@ export async function makeTrade(
 ): Promise<void> {
   const [trader] = await ethers.getSigners();
   const chain = ChainUtils.fromNetwork(network);
+  const api = new Api(network.name);
 
   console.log(`Using account ${trader.address}`);
 
   const allTokens = await fetchTokenList(tokenListUrl, chain);
-  const tokensWithBalance = await filterTokensWithBalance(
+  const tokensWithBalance = await filterTokensWithBalanceAndRoutesToBuyTokens(
```

Comment:
This name is a mouthful. Maybe:
```suggestion
  const tokensWithBalance = await filterTradableTokens(
```